src/dune_rules/pkg_rules.ml

```diff
@@ -702,7 +709,11 @@ module Run_with_path = struct
       let open Fiber.O in
       let display = !Clflags.display in
       match prog with
-      | Error e -> Action.Prog.Not_found.raise e
+      | Error e ->
+        let _, loc = pkg in
+        let depexts_msg = if List.is_empty depexts then [] else Depexts.message depexts in
+        let error_msg = (Action.Prog.Not_found.message e).paragraphs in
+        raise (User_error.E (error_msg @ depexts_msg |> User_message.make ~loc))
```

Comment:
It might be simpler to set the `hint` field of `e`. The benefit is that it would let us avoid adding new code to the engine.

Reply:
Is there a particular reason to not use `User_error.raise`? I feel it makes the code easier to mentally parse.

Reply:
The suggestion of @gridbugs was good one that simplifies the code.